src/StatusQ/Controls/StatusBaseInput.qml

```diff
@@ -0,0 +1,113 @@
+    property alias selectedTextColor: edit.selectedTextColor
+    property alias selectionStart: edit.selectionStart
+    property alias selectionEnd: edit.selectionEnd
+
+    property alias color: edit.color
+    property alias font: edit.font
+    property alias verticalAlignmet: edit.verticalAlignment
+    property alias horizontalAlignment: edit.horizontalAlignment
+
+    property alias placeholderText: placeholder.text
+    property alias placeholderTextColor: placeholder.color
+    property alias placeholderFont: placeholder.font
+
+    property real leftPadding: 16
+    property real rightPadding: 16
+    property real topPadding: 11
+    property real bottomPadding: 11
+
+    implicitWidth: 200
+    height: 40
+
+    clip: true
+
+    Rectangle {
+        anchors.fill: parent
+        color: Theme.palette.baseColor2
+        radius: 8
+    }
+
+    Flickable {
+        id: flick
+
+        anchors.fill: parent
+        anchors.leftMargin: statusBaseInput.leftPadding
+        anchors.rightMargin: statusBaseInput.rightPadding + clearButtton.width
+        anchors.topMargin: statusBaseInput.topPadding
+        anchors.bottomMargin: statusBaseInput.bottomPadding
+        contentWidth: edit.paintedWidth
+        contentHeight: edit.paintedHeight
+        clip: true
+
+
+        function ensureVisible(r)
+        {
+            if (contentX >= r.x)
+                contentX = r.x;
+            else if (contentX+width <= r.x+r.width)
+                contentX = r.x+r.width-width;
+            if (contentY >= r.y)
+                contentY = r.y;
+            else if (contentY+height <= r.y+r.height)
+                contentY = r.y+r.height-height;
+        }
+
+
+
+        TextEdit {
+            id: edit
+            width: flick.width
+            height: 22
+            selectByMouse: true
+            anchors.verticalCenter: parent.verticalCenter
+            focus: true
+
+            font.pixelSize: 15
+            font.family: Theme.palette.baseFont.name
+            color: Theme.palette.directColor1
+
+            onCursorRectangleChanged: flick.ensureVisible(cursorRectangle)
+            wrapMode: TextEdit.NoWrap
+
+            StatusBaseText {
+                id: placeholder
+                visible: edit.text.length === 0
+                anchors.left: parent.left
+                font.pixelSize: 15
+                font.family: Theme.palette.baseFont.name
+                color: Theme.palette.baseColor1
+            }
+        }
+    }
+
+    StatusFlatRoundButton {
+        id: clearButtton
+        visible: edit.text.length != 0
+        anchors.right: parent.right
+        anchors.rightMargin: 11
+        anchors.verticalCenter: parent.verticalCenter
+        type: StatusFlatRoundButton.Type.Secondary
+        width: 14
+        height: 14
+        icon.name: "clear"
+        icon.width: 14
+        icon.height: 14
+        onClicked: {
+            edit.clear()
+        }
+    }
```

Comment:
This clear button is nice, but I don't think we want this to always show in the status base input. Again, the base input should just be a simple input that can then be used to build more complex inputs.

I'm fine with making this configurable though. We could have a flag `clearable` which will render that button.

Also, we should double check the color of the clear button. Did you take it from any designed component?